SampleApp/Repository/EmployeeRepository.cs

```diff
@@ -0,0 +1,93 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using SampleApp.Data;
+using SampleApp.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SampleApp.Repository
+{
+    public class EmployeeRepository : IEmployeeRepository
+    {
+        private readonly EmployeeContext _context;
+        private readonly IMapper _mapper;
+
+        public EmployeeRepository(EmployeeContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        /// <summary>
+        /// Fetches the list of employees from the Employees table.
+        /// </summary>
+        /// <returns>List of employees.</returns>
+        public async Task<List<EmployeeModel>> GetAllEmployees(EmployeeQueryParameters parameters)
+        {
+            List<Employee> employees = await _context.Employees.Where(x =>
+                                                                    x.IsActive.Equals(parameters.Status) &&
+                                                                    (string.IsNullOrWhiteSpace(parameters.SearchString) ||
+                                                                    x.Name.Contains(parameters.SearchString) ||
+                                                                    x.Role.Contains(parameters.SearchString) ||
+                                                                    x.Salary.ToString().Contains(parameters.SearchString)) &&
+                                                                    (parameters.InitialDate == default || x.JoiningDate >= parameters.InitialDate) &&
+                                                                    (parameters.FinalDate == default || x.JoiningDate <= parameters.FinalDate)).ToListAsync();
+
+            employees = employees.Skip((parameters.CurrentPage - 1) * parameters.PageSize).Take(parameters.PageSize)?.ToList();
+            return _mapper.Map<List<EmployeeModel>>(employees);
+        }
+
+        /// <summary>
+        /// Fetches the employee model for the given Id.
+        /// </summary>
+        /// <param name="id">Id for the employee</param>
+        /// <returns>Employee model with the given Id.</returns>
+        public async Task<EmployeeModel> GetEmployeeById(Guid id)
+        {
+            Employee employee = await _context.Employees.AsNoTracking().Where(x => x.EmployeeGuid.Equals(id)).FirstOrDefaultAsync();
```

Comment:
consider using FirstOrDefaultAsync directly on the Employees DbSet instead of using AsNoTracking().Where(...). Since the method is performing a single lookup by EmployeeGuid, using FirstOrDefaultAsync directly would be more concise and efficient.

Reply:
Updated to use FirstOrDefaultAsync